shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -65,6 +73,24 @@ public TextInputPlugin(
       afm = null;
     }
 
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+      int mask = 0;
+      if ((View.SYSTEM_UI_FLAG_HIDE_NAVIGATION & mView.getWindowSystemUiVisibility()) == 0) {
+        mask = mask | WindowInsets.Type.navigationBars();
+      }
+      if ((View.SYSTEM_UI_FLAG_FULLSCREEN & mView.getWindowSystemUiVisibility()) == 0) {
+        mask = mask | WindowInsets.Type.statusBars();
+      }
+      imeSyncCallback =
+          new ImeSyncDeferringInsetsCallback(
+              view,
+              mask, // Overlay
```

Comment:
oh I see what this is after reading your other class constructor. Add comments to make this very clear. 

e.g. these are the things that'll reflect in flutter right away

these are the things that will animate